yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,583 @@
+
+        content_subdict = try_get(downloaded_json, lambda x: x['content'])
+        video_formats_url = try_get(content_subdict, lambda x: url_or_none(x['contentUrl']))
+        availability = try_get(self, lambda x: videoAvailability(x, downloaded_json))
+
+        if video_formats_url:
+            # Prior to adopting M3U, Rokfin stored videos directly in mp4 files:
+            if video_formats_url.endswith('.mp4'):
+                return self.url_result(url=video_formats_url, video_id=video_id, original_url=url_from_user)
+
+            if not(video_formats_url.endswith('.m3u8')):
+                self.raise_no_formats(msg=f'unsupported video URL {video_formats_url}', expected=False)
+
+            frmts = self._extract_m3u8_formats(m3u8_url=video_formats_url, video_id=video_id, fatal=False)
+            self._sort_formats(frmts)
+        else:
+            frmts = None
+
+            if availability == 'premium_only':
+                # The video is premium only.
+                self.raise_no_formats(msg='premium content', expected=True)
+            else:
+                # We don't know why there is no (valid) video URL present.
+                self.raise_no_formats(msg='unable to proceed due to missing meta data', expected=True)
+
+        downloaded_json = try_get(downloaded_json, lambda x: x.get)
+        created_by = try_get(downloaded_json, lambda x: x('createdBy')).get
+        upload_date_time = try_get(downloaded_json, lambda x: x('creationDateTime'))
+        channel_name = try_get(created_by, lambda x: x('name'))
+        content_subdict = try_get(content_subdict, lambda x: x.get)
+
+        return {
+            'id': video_id,
+            'url': video_formats_url,
+            'original_url': url_from_user,
+            'title': try_get(content_subdict, lambda x: x('contentTitle')),
+            'webpage_url': self._RECOMMENDED_VIDEO_BASE_URL + video_id,
+            'manifest_url': video_formats_url,
+            'is_live': False,
+            'was_live': False,
+            'duration': try_get(content_subdict, lambda x: float_or_none(x('duration'))),
+            'thumbnail': try_get(content_subdict, lambda x: x('thumbnailUrl1')),
+            'description': try_get(content_subdict, lambda x: x('contentDescription')),
+            'like_count': try_get(downloaded_json, lambda x: x('likeCount')),
+            'dislike_count': try_get(downloaded_json, lambda x: x('dislikeCount')),
+            'comment_count': try_get(downloaded_json, lambda x: x('numComments')),
+            'availability': availability,
+            'creator': channel_name,
+            'channel_id': try_get(created_by, lambda x: x('id')),
+            'channel': channel_name,
+            'channel_url': try_get(created_by, lambda x: self._CHANNEL_BASE_URL + x('username')),
+            'timestamp': unified_timestamp(upload_date_time),
+            'tags': [str(tag) for tag in try_get(downloaded_json, lambda x: x('tags')) or []],
```

Comment:
```suggestion
            'duration': content.get('duration'),
            'thumbnail': url_or_none(content.get('thumbnailUrl1')),
            'description': content.get('contentDescription'),
            'like_count': downloaded_json.get('likeCount'),
            'dislike_count': downloaded_json.get('dislikeCount'),
            'comment_count': downloaded_json.get('numComments'),
            'availability': availability,
            'creator': channel_name,
            'channel_id': try_get(downloaded_json, x['createdBy']['id']),
            'channel': channel_name,
            'channel_url': try_get(downloaded_json, self._CHANNEL_BASE_URL + x['createdBy']['username'])
            'timestamp': unified_timestamp(upload_date_time),
            'tags': [str(tag) for tag in downloaded_json.get('tags') or []],
```

Reply:
Also, `try_get(d, lambda x: x[a][b][c])` can be simplified to `traverse_obj(d, (a, b, c)` if you want